xgi/classes/hypergraph.py

```diff
@@ -1151,3 +1147,92 @@ def bunch_iter(nlist, nodes):
 
             bunch = bunch_iter(nbunch, self._node)
         return bunch
+
+    def max_edge_order(self):
+        """Returns the maximum order of edges in the hypergraph.
+
+        Returns
+        -------
+        d_max : int
+            Maximum order of edges in hypergraph
+        """
+
+        try:
+            edges = list(self._edge.values())
+            d_max = max([len(edge) for edge in edges]) - 1
+        except ValueError:  # if edges is empty
+            if len(self._node) > 0:
+                d_max = 0
+            else:
+                d_max = None
+
+        return d_max
+
+    def is_possible_order(self, d):
+        """Returns True if 'd' is a possible edge order.
+
+        Parameters
+        ----------
+        d : int
+            Order for which to check
+
+        Returns
+        -------
+        bool
+        """
+        d_max = self.max_edge_order()
+        return (d >= 1) and (d <= d_max)
+
+    def singleton_edges(self):
+        """Returns a dict of single edges"""
+
+        return {
+            id_: self._edge[id_]
+            for id_, size in dict(self.edge_size).items()
+            if size == 1
+        }
+
+    def remove_singleton_edges(self):
+        """Removes all singletons edges from the hypergraph"""
+
+        singleton_ids = [id_ for id_, size in dict(self.edge_size).items() if size == 1]
+        self.remove_edges_from(singleton_ids)
+        return None
+
+    def is_d_uniform(self, return_d=False):
```

Comment:
I'm not loving the design of this function. I think it'd be cleaner with a signature like the following:
```python
is_uniform(self)
```
and make it so that if the graph is uniform, it returns `d`, and if it is not uniform, it returns something that evaluates to false (like False, 0, or None). In this way, we always return `d` and we can still use it as a boolean check: `if H.is_uniform()` always works as expected.